src/main/java/com/influxdb/v3/client/internal/NanosecondConverter.java

```diff
@@ -111,4 +118,53 @@ public static BigInteger convert(final Instant instant, final WritePrecision pre
 
         return FROM_NANOS.get(precision).apply(nanos);
     }
+
+    /**
+     * Convert Long or LocalDateTime to timestamp nanosecond.
+     *
+     * @param value  the time in Long or LocalDateTime
+     * @param field the arrow field metadata
+     * @return the time in nanosecond
+     */
+    public static BigInteger getTimestampNano(@Nonnull final Object value, @Nonnull final Field field) {
+        BigInteger result = null;
+
+        if (value instanceof Long) {
+            if (field.getFieldType().getType() instanceof ArrowType.Timestamp) {
+                ArrowType.Timestamp type = (ArrowType.Timestamp) field.getFieldType().getType();
+                TimeUnit timeUnit;
+                switch (type.getUnit()) {
+                    case SECOND:
+                        timeUnit = TimeUnit.SECONDS;
+                        break;
+                    case MILLISECOND:
+                        timeUnit = TimeUnit.MILLISECONDS;
+                        break;
+                    case MICROSECOND:
+                        timeUnit = TimeUnit.MICROSECONDS;
+                        break;
+                    case NANOSECOND:
+                    default:
+                        timeUnit = TimeUnit.NANOSECONDS;
+                        break;
+                }
+                long nanoseconds = TimeUnit.NANOSECONDS.convert((Long) value, timeUnit);
+                Instant instant = Instant.ofEpochSecond(0, nanoseconds);
+                result = convertInstantToNano(instant);
+            } else {
+                Instant instant = Instant.ofEpochMilli((Long) value);
+                result = convertInstantToNano(instant);
+            }
+        } else if (value instanceof LocalDateTime) {
+            Instant instant = ((LocalDateTime) value).toInstant(ZoneOffset.UTC);
+            result = convertInstantToNano(instant);
+        }
+        return result;
+    }
+
+    private static BigInteger convertInstantToNano(final Instant instant) {
```

Comment:
Add `Nonnull`, ` Nullable` annotation:

```suggestion
    @Nullable
    private static BigInteger convertInstantToNano(@Nonnull final Instant instant) {
```